pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/metrics/LocMetric.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.oom.metrics;
+
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTMethodOrConstructorDeclaration;
+import net.sourceforge.pmd.lang.java.oom.ClassMetric;
+import net.sourceforge.pmd.lang.java.oom.OperationMetric;
+import net.sourceforge.pmd.lang.java.oom.visitor.PackageStats;
+
+/**
+ * Lines Of Code metric. Simply equates the length in lines of code of the measured entity.
+ *
+ * @author Clément Fournier
+ */
+public class LocMetric extends AbstractMetric implements ClassMetric, OperationMetric {
+
+    @Override
+    public double computeFor(ASTClassOrInterfaceDeclaration node, PackageStats holder) {
+        return node.getEndLine() - node.getBeginLine();
```

Comment:
This is a very simple method to calculate the lines. It depends on the source code formatting. I don't know about the requirements for this particular metric, but maybe NCSS (non-commenting sourcecode statements) might be an alternative. It of course it a bit more complex to calculate...